utPlayer.py

```diff
@@ -22,35 +22,63 @@
 #* ******************************************************************************
 import os
 import sys
-import subprocess
-import time
+from enum import Enum
 
 # Helper always exist in the same directory under raft and can use it for confirmation testing
 dir_path = os.path.dirname(os.path.realpath(__file__))
 sys.path.append(dir_path+"/../../../")
 
 from framework.core.logModule import logModule
 from framework.plugins.ut_raft.interactiveShell import InteractiveShell
+from framework.plugins.ut_raft.configRead import ConfigRead
+
+class MixerInputTypes(Enum):
+    MIXER_INPUT_PRIMARY     = 1
+    MIXER_INPUT_SECONDARY   = 2
+
 class utPlayer():
     """
     UT Player class
     """
-    def __init__(self, session:object, playerTool:dict={'tool': 'gstreamer', 'prerequisites':[]}, log:logModule=None):
+
+    def __init__(self, session:object, vendor:str, log:logModule=None):
         """
         Initializes player class.
 
         Args:
             session (class): The session object to communicate with the device
+            vendor (str): vendor name
             log (class, optional): Parent log class. Defaults to None.
         """
         self.log = log
         if log is None:
             self.log = logModule(self.__class__.__name__)
             self.log.setLevel( self.log.INFO )
+
         self.session = session
-        self.playbackTool = playerTool["tool"]
-        for cmd in playerTool["prerequisites"]:
-            self.session.write(cmd)
+
+        configPath = os.path.join(dir_path, "configs/utPlayerConfig.yml")
+        self.playerProfile = ConfigRead(configPath, vendor)
+        if self.playerProfile is None:
+            self.log.fatal("Profile not found[{}]".format(configPath))
+            sys.exit(1)
```

Comment:
for your info `self.log.fatal() - will do os.exit(1)`

Reply:
removed exit